astropy/utils/state.py

```diff
@@ -7,6 +7,44 @@
 __all__ = ['ScienceState']
 
 
+class _Context:
+    """
+    Science state context. Returned by :func:`~ScienceState.set`.
+
+    ::
```

Comment:
I don't find the example very helpful... Maybe just some text along the lines of
```
This is initialized and returned inside :meth:`~ScienceState.set`, so that one can
set the state in the context of a `with` statement.
```